File: 3146-permutation-difference-between-two-strings/3146-permutation-difference-between-two-strings.py
class Solution(object):
    def findPermutationDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: int
        """
        length=len(s)
        pd=0
        for i in range(length):
            for j in range(length):
                if(s[i]==t[j]):
                    pd+=abs(i-j)
        return pd

File: 3146-permutation-difference-between-two-strings/3146-permutation-difference-between-two-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var findPermutationDifference = function(s, t) {
    let m=new Map()
    let len=s.length
    for(let i=0;i<len;i++){
        m.set(s[i],i)
    }
    let pd=0;
    let x;
    for(let j=0;j<len;j++){
        pd+=Math.abs(m.get(t[j])-j)
    }
    return pd
};